### PR TITLE
If switch

### PR DESCRIPTION
**Pull request by limur88**
Homework 4

### CODE REVIEW

File: ifSwitchTernary/2_lectionAlertAllIn1.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    alert("Technical work")
+}
+
+//1.Реализовать Task 1 через switch
+const tooSmall = age < minAge;
+const good = age >= minAge && age < maxAge;
+const tooBig = age > maxAge;
+const outOfRange = 60;
+
+switch (tooSmall || good || tooBig || outOfRange) {
```
Comment (anatoly-karpovich) [inline ifSwitchTernary/2_lectionAlertAllIn1.js:38]: Interesting implementation!

File: ifSwitchTernary/task4.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// 3.Преобразовать Task 2 - 1 таким образом, чтобы значение НАПРИМЕР '2' (т.е. ЛЮБАЯ строка в которой лежат ТОЛЬКО ЦИФРЫ) пропускалось,
+// преобразовываясь в number
+const age = prompt("What is your age?");
+const minAge = 18;
+const maxAge = 60;
+
+const checkNumIsString = 0 + (+ age);
+
+if (typeof checkNumIsString != "number") {
```
Comment (anatoly-karpovich) [inline ifSwitchTernary/task4.js]: typeof NaN === 'number'
```suggestion
if (!isNaN(checkNumIsString)) {
```

File: ifSwitchTernary/task4.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+// 3.Преобразовать Task 2 - 1 таким образом, чтобы значение НАПРИМЕР '2' (т.е. ЛЮБАЯ строка в которой лежат ТОЛЬКО ЦИФРЫ) пропускалось,
+// преобразовываясь в number
+const age = prompt("What is your age?");
+const minAge = 18;
+const maxAge = 60;
+
+const checkNumIsString = 0 + (+ age);
```
Comment (anatoly-karpovich) [inline ifSwitchTernary/task4.js]: no need to add 0
if +age is NaN it will be still NaN

**Review by anatoly-karpovich — CHANGES_REQUESTED** [review]

File: ifSwitchTernary/2_lectionAlertAllIn1.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    }
+}
+
+// 2.Преобразовать написанный код в task 1 так, чтобы сначала проверялся тип данных.
+// И если он не number - кидалась ошибка в консоль.
+// Проверить работу кода на следующих данных 17, 18, 61, "2", "aaa"
+
+const checkNumIsString = + age;
+
+if (!checkNumIsString) {
```
Comment (VladimirLQA) [inline ifSwitchTernary/2_lectionAlertAllIn1.js]: Do not need to convert value, just check it's type
```suggestion

if (typeof age !== 'number') {
```

File: ifSwitchTernary/2_lectionAlertAllIn1.js [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+            alert("Technical work");
+            break;
+        }
+    }
+}
+
+// 3.Преобразовать Task 2 - 1 таким образом, чтобы значение НАПРИМЕР '2' (т.е. ЛЮБАЯ строка в которой лежат ТОЛЬКО ЦИФРЫ) пропускалось,
+// преобразовываясь в number
+
+if (!checkNumIsString) {
```
Comment (VladimirLQA) [inline ifSwitchTernary/2_lectionAlertAllIn1.js]:
```suggestion
if (isNaN(age)) {
```

File: ifSwitchTernary/2_lectionAlertAllIn1.js [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+    alert("Wrong type of data")
+} else {
+    const tooSmall = age < minAge;
```
Comment (VladimirLQA) [inline ifSwitchTernary/2_lectionAlertAllIn1.js:98]: In `else` block convert `age` value to number for next comparisons

**Review by VladimirLQA — DISMISSED** [review]

**Review by anatoly-karpovich — DISMISSED** [review]

**Review by shnobel — APPROVED** [review]

File: ifSwitchTernary/2_lectionAlertAllIn1.js (hunk 6 above)
Comment (limur88) [reply ifSwitchTernary/2_lectionAlertAllIn1.js:98]: age is converted to number as i use > /< for comparison, isn't it?

Comment (limur88) [reply ifSwitchTernary/2_lectionAlertAllIn1.js:98]: age is converted as i use >/<, isn't it?

Comment (VladimirLQA) [reply ifSwitchTernary/2_lectionAlertAllIn1.js:98]: Yea, you're right, but I see the `alert` key word, that means we get value from promt which returns a string, and we can only one time convert it to number and reuse

**Review by VladimirLQA — APPROVED** [review]

**Pull request merged**